guides/deployment/releases.md

```diff
@@ -189,26 +189,36 @@ ENV MIX_ENV=prod
 
 # install mix dependencies
 COPY mix.exs mix.lock ./
-COPY config config
-RUN mix do deps.get --only $MIX_ENV, deps.compile
+RUN mix deps.get --only $MIX_ENV
+RUN mkdir config
+# if our compile-time config changes, we may need to
+# recompile dependencies
+COPY config/config.exs config/prod.exs config/
+run mix deps.compile
 
 # build assets
 COPY assets/package.json assets/package-lock.json ./assets/
+# install all npm dependencies from scratch
 RUN npm --prefix ./assets ci --progress=false --no-audit --loglevel=error
 
 COPY priv priv
 COPY assets assets
+# use webpack to compile npm dependencies - https://www.npmjs.com/package/webpack-deploy
 RUN npm run --prefix ./assets deploy
 RUN mix phx.digest
 
-# compile and build release
+# compile and build the release
 COPY lib lib
```

Comment:
The only thing the be aware here is that the build will 'not work' if you you are purging your CSS (i.e., removing the unused CSS classes) because it will miss the source code from `lib/my_web`. However this is something that is already an issue with the current version of the `Dockerfile` but I just wanted to mention it here, maybe we should leave a comment about this in the `Dockerfile`?

Reply:
@dvic Sorry, I don't understand the problem yet. 😊 In an example project, if I delete some lines from `assets/css/somefile.css`, that busts the cache for the `COPY assets assets` step, which re-triggers the npm compilation step and should mean that the everything from there on is re-run. What would be broken in that case?

Reply:
No worries. I'll try to elaborate. The problem only occurs when you use something like https://purgecss.com. This plugin scans your source code and keeps track of all the "words" it encounters. It then discards any css classes that are not in this set of seen words (in order to minimize the css). The problem we encountered with the above Docker build is that `COPY lib lib` is being performed *after* `npm run deploy` has run, therefore, a library like `purgecss` won't see the Elixir source code and will miss all the classes used in the `.eex` templates.

Reply:
@dvic Got it, thanks for explaining. I added a comment.